src/v/ssx/auto_fmt.h

```diff
@@ -0,0 +1,103 @@
+                 << d << f6;
+    } else if constexpr (a == 8) {
+        auto& [f0, f1, f2, f3, f4, f5, f6, f7] = static_cast<const Derived&>(e);
+        return o << f0 << d << f1 << d << f2 << d << f3 << d << f4 << d << f5
+                 << d << f6 << d << f7;
+    } else if constexpr (a == 9) {
+        auto& [f0, f1, f2, f3, f4, f5, f6, f7, f8]
+          = static_cast<const Derived&>(e);
+        return o << f0 << d << f1 << d << f2 << d << f3 << d << f4 << d << f5
+                 << d << f6 << d << f7 << d << f8;
+    } else if constexpr (a == 10) {
+        auto& [f0, f1, f2, f3, f4, f5, f6, f7, f8, f9]
+          = static_cast<const Derived&>(e);
+        return o << f0 << d << f1 << d << f2 << d << f3 << d << f4 << d << f5
+                 << d << f6 << d << f7 << d << f8 << d << f9;
+    } else if constexpr (a == 11) {
+        auto& [f0, f1, f2, f3, f4, f5, f6, f7, f8, f9, f10]
+          = static_cast<const Derived&>(e);
+        return o << f0 << d << f1 << d << f2 << d << f3 << d << f4 << d << f5
+                 << d << f6 << d << f7 << d << f8 << d << f9 << d << f10;
+    } else if constexpr (a == 12) {
+        auto& [f0, f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11]
+          = static_cast<const Derived&>(e);
+        return o << f0 << d << f1 << d << f2 << d << f3 << d << f4 << d << f5
+                 << d << f6 << d << f7 << d << f8 << d << f9 << d << f10 << d
+                 << f11;
+    } else if constexpr (a == 13) {
+        auto& [f0, f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11, f12]
+          = static_cast<const Derived&>(e);
+        return o << f0 << d << f1 << d << f2 << d << f3 << d << f4 << d << f5
+                 << d << f6 << d << f7 << d << f8 << d << f9 << d << f10 << d
+                 << f11 << d << f12;
+    } else if constexpr (a == 14) {
+        auto& [f0, f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11, f12, f13]
+          = static_cast<const Derived&>(e);
+        return o << f0 << d << f1 << d << f2 << d << f3 << d << f4 << d << f5
+                 << d << f6 << d << f7 << d << f8 << d << f9 << d << f10 << d
+                 << f11 << d << f12 << d << f13;
+    } else if constexpr (a == 15) {
+        auto& [f0, f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11, f12, f13, f14]
+          = static_cast<const Derived&>(e);
+        return o << f0 << d << f1 << d << f2 << d << f3 << d << f4 << d << f5
+                 << d << f6 << d << f7 << d << f8 << d << f9 << d << f10 << d
+                 << f11 << d << f12 << d << f13 << d << f14;
+    } else if constexpr (a == 16) {
+        auto& [f0, f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11, f12, f13, f14, f15]
+          = static_cast<const Derived&>(e);
+        return o << f0 << d << f1 << d << f2 << d << f3 << d << f4 << d << f5
+                 << d << f6 << d << f7 << d << f8 << d << f9 << d << f10 << d
+                 << f11 << d << f12 << d << f13 << d << f14 << d << f15;
+    }
+    return o;
```

Comment:
really love the idea, may i suggest this?
https://gcc.godbolt.org/z/ex3Gf6z9s

it accepts the strings and relies on reflection::to_tuple to export the fields, so that if we need to support more fields (hello topic_properties) we just need to adjust reflection to tuple)

(still have to read the rest)